packages/sqlite_async/lib/src/common/sqlite_database.dart

```diff
@@ -38,6 +38,21 @@ mixin SqliteDatabaseMixin implements SqliteConnection, SqliteQueries {
   ///
   /// Use this to access the database in background isolates.
   IsolateConnectionFactory isolateConnectionFactory();
+
+  /// TODO Improve on this definition by supporting a writeable context.
+  Future<void> exclusiveLock<T>(
+    Future<T> Function(SqliteReadContext ctx) callback,
+  ) {
```

Comment:
The return value of the callback function is never used - rather remove the generic `<T>` and let the callback return `Future<void>`.